jellyfin-api/src/main/kotlin/org/jellyfin/apiclient/api/client/Response.kt

```diff
@@ -1,12 +1,81 @@
 package org.jellyfin.apiclient.api.client
 
+import io.ktor.http.*
 import kotlin.reflect.KProperty
 
+/**
+ * Response from a HTTP class in the [ApiClient].
+ *
+ * @param status - See [HttpStatusCode]
+ */
 class Response<T>(
-	private val data: T
+	val content: T,
+	val status: Int,
+	val headers: Map<String, List<String>>
 ) {
+	/**
+	 * Get the response content using property delegation.
+	 *
+	 * ```kt
+	 * val content by response
+	 * ```
+	 */
 	@JvmSynthetic
-	operator fun getValue(thisRef: Any?, property: KProperty<*>): T = data
+	operator fun getValue(thisRef: Any?, property: KProperty<*>): T = content
 
-	fun getData() = data
+	/**
+	 * Check if the HTTP response status is equal to a given status.
+	 *
+	 * ```kt
+	 * val isOk = response isStatus 200
+	 * val isOk = response.isStatus(200)
+	 * ```
+	 */
+	infix fun isStatus(comparable: Int) = status == comparable
+
+	/**
+	 * Check if the HTTP response status is not equal to a given status.
+	 *
+	 * ```kt
+	 * val isNotOk = response isNotStatus 200
+	 * val isNotOk = response.isNotStatus(200)
+	 * ```
+	 */
+	infix fun isNotStatus(comparable: Int) = status != comparable
+
+	/**
+	 * Check if the HTTP response status is in a given range.
+	 *
+	 * ```kt
+	 * val isOk = response isStatusIn (200 until 300)
+	 * val isOk = response.isStatusIn(IntRange(200, 299))
+	 * ```
+	 */
+	infix fun isStatusIn(comparable: IntRange) = status in comparable
+
+	/**
+	 * Check if the HTTP response status is in the 2xx range.
+	 */
+	fun isStatus200() = this isStatusIn (200 until 300)
+
+	/**
+	 * Check if the HTTP response status is in the 4xx range.
+	 */
+	fun isStatus400() = this isStatusIn (400 until 500)
+
+	/**
+	 * Check if the HTTP response status is in the 5xx range.
+	 */
+	fun isStatus500() = this isStatusIn (500 until 600)
```

Comment:
Are those really that necessary? I'm not sure of the benefit these functions bring..

Reply:
I've actually used `response isStatus 200` in an upcoming PR

Reply:
But is that actually so much better than a `response.status == 200`?

Reply:
Hmm.. actually you're right. The status value was private at first but since it's public anyway we don't really need these functions.